ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/dialogs/PluginSelectionDialog.java

```diff
@@ -153,8 +159,8 @@ private static IPluginModelBase[] getElements(boolean includeFragments) {
 		return PluginRegistry.getActiveModels(includeFragments);
 	}
 
-	public static HashSet<String> getExistingImports(IPluginModelBase model, boolean includeImportPkg) {
-		HashSet<String> existingImports = new HashSet<>();
+	public static HashMap<String, Boolean> getExistingImports(IPluginModelBase model, boolean includeImportPkg) {
```

Comment:
Please use only collection interfaces (as abstract as possible) if possible here and in all other changed code. Using a specific implementation is usually unnecessarily restrictive.
```suggestion
	public static Map<String, Boolean> getExistingImports(IPluginModelBase model, boolean includeImportPkg) {
```